README.md

```diff
@@ -2,6 +2,118 @@
 
 Option is a C# representation of the maybe monad
 
+### Monad
+Monads are used to encapsulate a value and simplify handling of unusual cases. You can think of monads as generic types.
+
+### Maybe
+Maybe is a data type (also known as an option type) which is used to represent an optional value.
+
+### Maybe Monad
+Maybe monad is a generic type which contains either some value of it's underlying type or nothing.
+
+If you need more information you can check [wikibooks introduction](https://en.wikibooks.org/wiki/Haskell/Understanding_monads/Maybe) to maybe monad.
```

Comment:
Consider removing the "If you need" text and instead attach the hyperlink to the words "Maybe monad".